tests/unit/config/test_qtargs.py

```diff
@@ -428,6 +428,51 @@ def test_env_vars(self, monkeypatch, config_stub,
 
         assert os.environ[envvar] == expected
 
+    @pytest.mark.parametrize('init_val, config_val', [
+        (   # Test setting a variable
+            {'QT_SCALE_FACTOR': None},
+            {'QT_SCALE_FACTOR': '2'},
+        ),
+        (   # Test changing a set variable
+            {'QT_SCALE_FACTOR': '2'},
+            {'QT_SCALE_FACTOR': '4'},
+        ),
+        (   # Test setting an unset variable
+            {'QT_SCALE_FACTOR': 'unset'},
+            {'QT_SCALE_FACTOR': '3'},
+        ),
+        (   # Test unsetting a variable which is set
+            {'QT_SCALE_FACTOR': '3'},
+            {'QT_SCALE_FACTOR': None},
+        ),
+        (   # Test unsetting a variable which is unset
+            {'QT_SCALE_FACTOR': 'unset'},
+            {'QT_SCALE_FACTOR': None},
+        ),
+        (   # Test setting multiple variables
+            {'QT_SCALE_FACTOR': None, 'QT_PLUGIN_PATH': None, 'QT_NEWVAR': None},
+            {'QT_SCALE_FACTOR': '3', 'QT_PLUGIN_PATH': '/tmp/', 'QT_NEWVAR': 'newval'},
+        )
+    ])
+    def test_environ_settings(self, monkeypatch, config_stub,
+                              init_val, config_val):
+        """Test setting environment variables using qt.environ."""
+        for var, val in init_val.items():
+            if val is not None:
+                if val == 'unset':
+                    monkeypatch.delenv(var)
```

Comment:
```suggestion
                    monkeypatch.delenv(var, raising=False)
```

so this doesn't fail when the environment variable to be deleted doesn't exist.